portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -16,22 +16,20 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
 
 /** Servlet that deletes some data. */
 @WebServlet("/delete-data")
 public class DeleteServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      
-      Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      PreparedQuery results = datastore.prepare(query);
+      long id = Long.parseLong(request.getParameter("comment-id"));
+      Key key = KeyFactory.createKey("Comment", id);
+      datastore.delete(key);
 
-      ArrayList<Entity> entities = new ArrayList<>();
-      for (Entity entity : results.asIterable()) {
-          datastore.delete(entity.getKey());
-      }
```

Comment:
I haven't seen anyone try using the DELETE verb? Can you try doing this in the DataServlet with `doDelete`?

Also I think you probably only want to be able to delete something that you created, right?

Reply:
yep, so the delete button only shows up for the person who wrote the comment which is handled in the javascript

Reply:
so i fixed it so that it checks if the user is the same on the server side as well, but i looked it up and it seems as though you can't use a DELETE method with an HTML form so i don't know how i could use doDelete https://stackoverflow.com/questions/33074877/why-a-browser-only-supports-get-and-post-http-methods

Reply:
A bit disappointing, but interesting.